cli/loader_lib.c

```diff
@@ -153,6 +154,82 @@ JL_DLLEXPORT const char * jl_get_libdir()
     return lib_dir;
 }
 
+#ifdef LIBSTDCXXPROBE
+// Write the path to the libstdcxx to load to "path", which should be a buffer of at least
+// JL_PATH_MAX, and the length of that string (excluding null terminator) to "written"
+// Returns 1 on success or 0 on error, in which case the caller should abort the process.
+#define JL_DEFAULT_STDCXXLIB "libstdc++.so.6"
+static int libstdcxxprobe(char* path, size_t* written) {
+    // On Linux, it can happen that the system as a newer libstdc++ than the one we ship,
+    // which can break loading of some system libraries:
+    // <https://github.com/JuliaLang/julia/issues/34276>.
+    int ret;
+    pid_t pid;
+    int fork_pipe[2];
+    void *handle;
+    char *dlerr;
+
+    // Create the pipe and child process.
+    ret = pipe(fork_pipe);
+    if (ret == -1) return 0;
+    pid = fork();
+    if (pid == -1) return 0;
+
+    if (pid == (pid_t) 0) { // Child process.
+        ret = close(fork_pipe[0]);
+        if (ret == -1) exit(1);
```

Comment:
The child process must use `_exit(0)`
```suggestion
        if (ret == -1) _exit(1);
```